src/CopyrighFixer/CopyrighFixer/signer.cpp

```diff
@@ -5,27 +5,151 @@
+
+            if (!currFM->write(i_file.filePath(), resultSign)) {
+                QuasarAppUtils::Params::log(QString("Failed to write signature in file - %1").arg(i_file.fileName()),
+                                            QuasarAppUtils::VerboseLvl::Error);
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
 bool Signer::checkSign(const Config &objConf) {
+
+    if (!processSign(objConf.getSrcDir(), objConf)) {
+        return false;
+    }
+
     return true;
 }
 
-IFileManager *Signer::searchFileByExt(const QString &extension) {
+const Signature Signer::upgradeOwner(const Signature &signConf, const Signature &fileSign) const {
+
+    int unixTime = time(0);
+
+    Signature signForSing = fileSign;
+    QMap<int, CopyrighFixer::Owner> mapOwnersFile = signForSing.getMapOwn();
+
+    CopyrighFixer::Owner newOwner = mapOwnersFile.take(fileSign.getMapOwn().cbegin().key());
+    newOwner.setName(signConf.getMapOwn().cbegin().value().getOwnerName());
+    newOwner.setTimePoint(unixTime);
+
+    mapOwnersFile.insert(unixTime, newOwner);
+    signForSing.setMapOwners(mapOwnersFile);
+
+    return signForSing;
+
+}
+
+const Signature Signer::appendOwner(const Signature &signConf, const Signature &fileSign) const {
+
+    int unixTime = time(0);
+
+    Signature signForSing = fileSign;
+    QMap<int, CopyrighFixer::Owner> mapOwnersFile = signForSing.getMapOwn();
+
+    CopyrighFixer::Owner newOwners;
+    newOwners.setName(signConf.getMapOwn().cbegin().value().getOwnerName());
+    newOwners.setTimePoint(unixTime);
+
+    mapOwnersFile.insert(unixTime, newOwners);
+    signForSing.setMapOwners(mapOwnersFile);
+
+    return signForSing;
+}
+
+const Signature Signer::mergeSign(const Signature &userSign, const Signature &fileSign) const {
+
+    bool isSizeOwnsEqual = userSign.getMapOwn().cbegin().value().getOwnerName() != fileSign.getMapOwn().cbegin().value().getOwnerName();
+    bool isSizeOwnsDiff = userSign.getMapOwn().cbegin().value().getOwnerName() == fileSign.getMapOwn().cbegin().value().getOwnerName();
+    bool isNumOwnsMore = userSign.getMapOwn().size() > 1;
+    bool isEqualLicTitle = userSign.getLicenseTitle() != fileSign.getLicenseTitle();
+
+    if (!fileSign.isValid()) {
+        return userSign;
+    }
+
+    if (isEqualLicTitle) {
+        QuasarAppUtils::Params::log("The signature in the file is different from the config signature.",
+                                    QuasarAppUtils::VerboseLvl::Warning);
+        return fileSign;
+    }
+
+    if (isNumOwnsMore) {
+        QuasarAppUtils::Params::log("Config signature contains more owners.",
+                                    QuasarAppUtils::VerboseLvl::Warning);
+
+        if (isSizeOwnsEqual) {
+            return appendOwner(userSign, fileSign);
+        }
+    }
+
+
+    if (isSizeOwnsDiff) {
+        return upgradeOwner(userSign, fileSign);
+    }
+
+    if (isSizeOwnsEqual) {
+        return appendOwner(userSign, fileSign);
+    }
```

Comment:
```suggestion
    bool fLastOwnerEqual = userSign.getMapOwn().cbegin().value().getOwnerName() != fileSign.getMapOwn().cbegin().value().getOwnerName();
    bool fLastOwnerDiff = userSign.getMapOwn().cbegin().value().getOwnerName() == fileSign.getMapOwn().cbegin().value().getOwnerName();
    bool fNumOwnsMore = userSign.getMapOwn().size() > 1;
    bool fEqualLicTitle = userSign.getLicenseTitle() != fileSign.getLicenseTitle();

    if (!fileSign.isValid()) {
        return userSign;
    }

    if (fEqualLicTitle) {
        QuasarAppUtils::Params::log("The signature in the file is different from the config signature.",
                                    QuasarAppUtils::VerboseLvl::Warning);
        return fileSign;
    }

    if (fNumOwnsMore) {
        QuasarAppUtils::Params::log("Config signature contains more owners.",
                                    QuasarAppUtils::VerboseLvl::Warning);

        if (fLastOwnerEqual) {
            return appendOwner(userSign, fileSign);
        }
    }


    if (fLastOwnerDiff) {
        return upgradeOwner(userSign, fileSign);
    }

    if (fLastOwnerEqual) {
        return appendOwner(userSign, fileSign);
    }
```

## About variables names:
* **is** - this prefix use for the method and functions that check something.  
* **f** - this prefix use for boolean variables. f ~~ if 
* i replaced your "sizeOvners" to "last owner". because you do not check size, you check last owner only